middleman_protocol/tests/test_helpers.py

```diff
@@ -0,0 +1,76 @@
+from asyncio import StreamReader
+
+import pytest
+from assertpy import assert_that
+from mock import MagicMock
+from mock import Mock
+
+from middleman_protocol import constants
+from middleman_protocol import exceptions
+from middleman_protocol.constants import ESCAPE_CHARACTER
+from middleman_protocol.constants import ESCAPE_SEQUENCES
+from middleman_protocol.constants import FRAME_SEPARATOR
+from middleman_protocol.helpers import handle_frame_receive_async, map_exception_to_error_code
+
+SOME_BYTES = b'1234567890qwerty'
+
+MESSAGE = SOME_BYTES + FRAME_SEPARATOR
+
+MESSAGE_WITH_ESCAPE_CHARACTER_SEQUENCE = SOME_BYTES + ESCAPE_SEQUENCES[ESCAPE_CHARACTER] + FRAME_SEPARATOR
+
+
+def async_stream_reader_mock(*args, **kwargs):
+    m = MagicMock(*args, **kwargs)
+
+    async def mock_coro(*a, **kw):
+        return m(*a, **kw)
+
+    mock_coro.mock = m
+    return mock_coro
+
+
+class TestHandleFrameReceiveAsync:
+    def test_that_when_unescaped_data_with_separator_is_received_unescaped_data_is_returned(self, event_loop):
```

Comment:
The received data is always escaped. Which also can mean that none of the bytes was replaced.

Reply:
renamed; done